src/terms/xc.jl

```diff
@@ -51,6 +55,9 @@ end
     n_spin   = model.n_spin_components
     @assert all(family(xc) in (:lda, :gga, :mgga, :mggal) for xc in term.functionals)
 
+    # Add the non-linear core correction density
+    ρ .+= term.ρ_nlcc
```

Comment:
This is modifying the density in the caller, you need a `rho = copy(rho)` or similar before

Reply:
Ahh, good catch!

Reply:
Also be careful that in the case of spin this might add rho_nlcc to both the spin up and down channels, whereas you probably need to add half?

Reply:
Also yes; do the spin channels come in together as rho, or are the evaluated separately?

Reply:
Rho is four dimensional, the last dimension is the spin channel. See total_density and spin_density

Reply:
Ok, still not sure that I'm handling this correctly, but I think it should at least be better now.

Reply:
Don't think so : spin density is spin up minus spin down, so probably zero?

Reply:
Yes 🤦. I should really read the docs before doing stupid stuff.

In that case, I think it should always just be `p_from_total_and_spin(ptot, nothing)` where `ptot` is the superposition of core charge densities.